ARM: Introduce Infineon PSoC 6 SoC

npal-cy opened the pull request:
This introduce an Infineon PSoC 6 SoC (https://github.com/zephyrproject-rtos/zephyr/issues/42883). 
This is a first MR which include following part:
- [Infineon PSoC 6 SOC Devicetree](https://github.com/zephyrproject-rtos/zephyr/commit/caa65aefca71222d7e1a81ff57a8a39f33a10c55)
- [Infineon CAT1/PSoC 6 SOC integration](https://github.com/zephyrproject-rtos/zephyr/commit/bbe2caadab2da5517c1588fc0ed1b3c6564a7a47)
- [CAT1 UART driver](https://github.com/zephyrproject-rtos/zephyr/commit/45d9de13f7dce9006c460dc8b6e70d0c6011da02) / [binding for uart](https://github.com/zephyrproject-rtos/zephyr/commit/98891b2510c458206d7d792370eeaf971f816689)
- [CAT1 GPIO driver](https://github.com/zephyrproject-rtos/zephyr/commit/89d61c120c6b6539d03b4b239a5af0b4ece57613) / [binding for gpio](https://github.com/zephyrproject-rtos/zephyr/commit/aa8af3e30e1d70b6f4505c8ac24282e9bdf7068b)
- [CAT1 Pin controller driver](https://github.com/zephyrproject-rtos/zephyr/commit/2b33f3d81831b264a3c2c53e450341040fb2da1b) / [binding for pinctrl](https://github.com/zephyrproject-rtos/zephyr/commit/dbe149b2aa837d11fa8462423b7105e394b99fee)

The next PRs will include Counter driver, CYW42XXX BT HCI extension driver, CYW42XXX wifi driver, etc.

PSoC 6 Integration structure:
![image](https://user-images.githubusercontent.com/46566584/160122509-cd6f4982-6545-451d-b214-710bf270cac9.png)

This PR has addressed comments from previous draft: https://github.com/zephyrproject-rtos/zephyr/pull/42725

To decrease number of files for review the tds and soc has files only for one MPN (CY8C624ABZI_S2D44), and die family (psoc6_02):
- [dts: arm: Introduce Infineon PSoC 6 SOC Devicetree](https://github.com/zephyrproject-rtos/zephyr/commit/caa65aefca71222d7e1a81ff57a8a39f33a10c55)
- [soc: arm: Introduce Infineon CAT1/PSoC 6 SOC integration](https://github.com/zephyrproject-rtos/zephyr/commit/bbe2caadab2da5517c1588fc0ed1b3c6564a7a47)

Rest of MPNs, die families will be added in separate PR .

npal-cy committed:
CODEOWNERS: Update codeowners for Infineon/Cypress sources

- Add Ian Fyall (@ifyall) as codeowner for Infineon/Cypress sources.
- Add Nazar Palamar (@npal-cy) as codeowner for Infineon/Cypress sources.
- Remove Gerson Fernando Budke (@nandojve) from codeowner
of Cypress sources.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy committed:
dts: arm: Introduce Infineon PSoC 6 SOC Devicetree

Added initial version of Devicetree for Infineon PSoC 6 SOC with following
structure:
 1. MPN devicetree files
  |--> psoc6
    |--> mpns
         |--> CY8C6016BZI_F04.dtsi
         |--> CY8C6036BZI_F04.dtsi
         |--> CY****.dtsi

  Those file describes cpus, flash-controller, sram memory, nvic option. It
  includes the package dtsi (e.g. psoc6_02.124-bga.dtsi) with information
  about gpio (based on package e.g. 68-qfn, 128-tqfp, 124-bga, etc.) and
  peripherals for (based on PSoC 6 series, psoc6_01, psoc6_02, etc).

  MPN devicetree file is main platform dtsi file, which should be included
  from board dts (e.g cy8cproto_062_4343w.dts), example:
  #include <infineon/psoc6/mpns/CY8C624ABZI_S2D44.dtsi>

 2. Devicetree files for PSoC 6 series 02 (2M).
  Includes: psoc6_02.dtsi - peripherals dtsi psoc6_01.xxxxx.dtsi - package
  dtsi. User does not directly include those files.
  It automatically includes via MPN dtsi.
   |--> psoc6_02
         |--> psoc6_02.dtsi
         |--> psoc6_02.100-wlcsp.dtsi
         |--> psoc6_02.124-bga.dtsi
         |--> psoc6_02.128-tqfp.dtsi
         |--> psoc6_02.68-qfn.dtsi

  In future PR/commits will be added Devicetree for support all
  PSoC 6 series:
   - for PSoC 6 series 01 (1M)
   - for PSoC 6 series 03 (512)
   - for PSoC 6 series 04 (256)

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy committed:
soc: arm: Introduce Infineon CAT1/PSoC 6 SOC integration

Add initial version of Infineon CAT1/PSoC 6 SOC integration.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy committed:
drivers: serial: Add Infineon CAT1 UART driver

Added initial version of Infineon CAT1 UART driver.
Added initial version of binding file for Infineon CAT1 UART driver.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy committed:
drivers: gpio: Add Infineon CAT1 GPIO driver

Added initial version of Infineon CAT1 GPIO driver.
Added initial version of binding file for Infineon CAT1 GPIO driver.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy commented:
Now compliance checks is clear. We have one warning, which is expected:
```
197cce89cd4a6ea3740ddc37f0edb00c0747702b:158: WARNING:NEW_TYPEDEFS: do not add new typedefs
#158: FILE: soc/arm/infineon_cat1/common/pinctrl_soc.h:48:
+typedef struct {
```
pinctrl requires to have typedef for pinctrl_soc_pin_t

npal-cy committed:
boards: arm: Introduce Infineon CY8CPROTO-062-4343W board

Add initial version of CY8CPROTO-062-4343W board.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy commented:
Added clock control driver and Infineon CY8CPROTO-062-4343W board. This is all content for this PR.

carlescufi commented:
@nandojve @gmarull @MaureenHelm @henrikbrixandersen reviews appreciated.

npal-cy committed:
modules: hal_infineon: Added mtb-hal-cat1 asset for Infenion Cat1 devices

- Add Cmakefiles for mtb-hal-cat1
- Update top makefile

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

carlescufi commented on soc/arm/infineon_cat1/psoc6/CMakeLists.txt:
```diff
@@ -0,0 +1,21 @@
+zephyr_sources(soc.c)
+zephyr_include_directories(.)
+
+include(${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.psoc6_02.txt)
```
I am not sure if this is good practice. @tejlmand

npal-cy replied:
i have die family specific CMakeLists.psoc6_xx.txt files (it was grouped in such way for easy maintains/update in future)
CMakeLists.psoc6_01.txt
CMakeLists.psoc6_02.txt (In this PR we add only this one)
CMakeLists.psoc6_03.txt
CMakeLists.psoc6_04.txt

So you suggest to put all content from CMakeLists.psoc6_xx.txt to CMakeLists.txt to get rid of "include".
Is it because consistency reason, or there are some specific of using "include" in Cmake at all...?

carlescufi commented on soc/arm/infineon_cat1/psoc6/CMakeLists.txt:
```diff
@@ -0,0 +1,21 @@
+include(${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.psoc6_02.txt)
+
+# PSoC 6 family defines
+zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_CAT1A COMPONENT_CAT1A)
```
Again, not sure if this belongs here or in the `modules/` folder

nandojve replied:
I have same feeling....

npal-cy replied:
```
# PSoC 6 family defines
zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1A COMPONENT_CAT1A)
zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1 COMPONENT_CAT1)
zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1 CY_USING_HAL)
zephyr_compile_definitions_ifdef(CONFIG_SOC_FAMILY_INFINEON_CAT1 ${CONFIG_SOC})
```
moved to modules/hal_infineon

carlescufi commented on drivers/serial/CMakeLists.txt:
```diff
@@ -47,6 +47,7 @@ zephyr_library_sources_ifdef(CONFIG_UART_NEORV32 uart_neorv32.c)
 zephyr_library_sources_ifdef(CONFIG_USART_GD32 usart_gd32.c)
 zephyr_library_sources_ifdef(CONFIG_UART_XEN_HVC uart_hvc_xen.c)
 zephyr_library_sources_ifdef(CONFIG_UART_XEN_HVC_CONSOLEIO uart_hvc_xen_consoleio.c)
+zephyr_library_sources_ifdef(CONFIG_UART_INFINEON_CAT1 uart_infineon_cat1.c)
```
Looking at this I wonder if we should use some sort of abbreviation, like `uart_inf_cat1.c`, or do you have any other preference?

ifyall replied:
We will use uart_ifx_cat1.c as the pattern.

npal-cy replied:
updated

carlescufi commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+						uint8_t signal_name);
+
+/* Helper API */
+static cyhal_uart_parity_t _convert_uart_parity_z_to_cyhal(uint8_t z_parity)
```
Please don't use `z_` as a prefix. This is reserved for Zephyr internal variables/functions. I suggest instead using `cy_parity` in the other variable.

ifyall replied:
If we need a prefix here, ideally we would use ifx or mtb. We will avoid usage of CY as a prefix.

npal-cy replied:
updated

carlescufi commented on dts/bindings/gpio/infineon,mtbhal-gpio.yaml:
```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2020-2021 ATL Electronics
+# SPDX-License-Identifier: Apache-2.0
+
+description: Infineon CAT1 GPIO PORT node
```
should this perhaps be squashed with the previous commit that introduces the GPIO driver?

ifyall replied:
agreed

npal-cy replied:
squashed

carlescufi commented on dts/bindings/pinctrl/infineon,mtbhal-pinctrl.yaml:
```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
```
Also here, squash with previous commit?

npal-cy replied:
squashed

carlescufi commented on include/zephyr/dt-bindings/pinctrl/cat1-pinctrl.h:
```diff
@@ -0,0 +1,253 @@
+/* Copyright 2022 Cypress Semiconductor Corporation (an Infineon company) or
```
Same, squash with previous commit?

carlescufi commented on dts/arm/infineon/psoc6/system_clocks.dtsi:
```diff
@@ -0,0 +1,144 @@
+/*
```
@erwango and @galak would be great if you could take a look at this.

carlescufi commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,205 @@
+OpenOCD Installation
+====================
+
+You must download `Infineon OpenOCD`_ from Github to flash and debug.
```
@stephanosio should we consider pulling this into the SDK?

carlescufi commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,205 @@
+
+    west init
+
+2. Add an infineon_connectivity.yaml manifest file in to zephyr/submanifests folder. There is an example yaml file in this directory.
```
That's a very good solution for the time being. @sylvioalves FYI.

mbolivar-nordic replied:
Wait, is this just a temporary place for openocd support to live? If so, OK. If this is to sneak in binary blobs, then no way. Please confirm.

carlescufi replied:
It seems that this is for binary blobs, so we will need to address this once and for all.

npal-cy replied:
correct it is binary blobs (now it lives https://github.com/Infineon/infineon_zephyr_connectivity.git, finally we planed to moved it to https://github.com/Infineon/infineon_zephyr_connectivity.git).

Now i understood that we should align with https://github.com/zephyrproject-rtos/zephyr/pull/44211#pullrequestreview-1001597600

carlescufi commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,205 @@
+
+.. code-block:: console
+
+   west flash --openocd /path/to/your/openocd/bin/openocd.exe
```
I would suggest you instead use something like `path/to/infineon/openocd/bin/openocd.exe`

npal-cy replied:
done

carlescufi reviewed (dismissed):
Thanks @npal-cy for the PR! This looks pretty good already! I have added some suggestions, questions and comments.

carlescufi commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,205 @@
+   manifest:
+     projects:
+       - name: infineon_zephyr_connectivity
+          url: https://github.com/Infineon/infineon_zephyr_connectivity.git
```
Is this link functional?

npal-cy replied:
not for now.

for now you can use: https://github.com/ifyall/infineon_zephyr_connectivity
we had plan for https://github.com/Infineon/infineon_zephyr_connectivity.git make public, in scope of this PR..

mbolivar-nordic reviewed (dismissed):
Please feel free to dismiss this review if your submanifest is just a temporary workaround to pull in an openocd build.

If on the other hand this is pulling in binary blobs, then no, we cannot merge it like this. And it will finally be time for someone to step up to:

1. design and implement the west extension that the TSC voted must be created to manage binary blobs
2. use the process defined in the document associated with that vote to propose the blobs be added
3. work with espressif to migrate the existing blobs we've hacked in with an SoC-specific west extension to the generic solution

Please read this document for a detailed discussion of the above points:

https://docs.google.com/document/d/1heqcv7dzGvM5rA9xpTMW3kyKJLpZsl2Gjsmr5eqBje8/edit

Results of the vote on this document are available in the TSC meeting minutes from feb. 9:

https://docs.google.com/document/d/1_sFM1BTLHrwRA9pbNoD7XPHE3OuqXT0P04AlAS2dvTk/edit#heading=h.qaca56pzh1r5

gmarull commented on dts/arm/infineon/psoc6/mpns/CY8C624ABZI_S2D44.dtsi:
```diff
@@ -0,0 +1,59 @@
+		reg = < 0x40240000 0x10000 >;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		label = "CYPRESS_FLASH_DRV_NAME";
```
why `_DRV_NAME` suffix?

npal-cy replied:
good catch. need to be update to something like "PSOC6_FLASH_CONTROLLER"

npal-cy replied:
updated

gmarull commented on dts/arm/infineon/psoc6/mpns/CY8C624ABZI_S2D44.dtsi:
```diff
@@ -0,0 +1,59 @@
+	};
+
+};
+
+
+
+&nvic {
```
1 empty line is sufficient

gmarull commented on dts/arm/infineon/psoc6/psoc6_02/psoc6_02.124-bga.dtsi:
```diff
@@ -0,0 +1,2051 @@
+
+#include <dt-bindings/gpio/gpio.h>
+#include "psoc6_02.dtsi"
+#include "dt-bindings/pinctrl/cat1-pinctrl.h"
```
please don't use `""` for non-local includes

npal-cy replied:
updated

gmarull commented on dts/arm/infineon/psoc6/psoc6_02/psoc6_02.124-bga.dtsi:
```diff
@@ -0,0 +1,2051 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/gpio/gpio.h>
```
this is missing `<zephyr/` prefix (check others as well)

npal-cy replied:
done

carlescufi commented:
> * design and implement the west extension that the TSC voted must be created to manage binary blobs
> * use the process defined in the document associated with that vote to propose the blobs be added
> * work with espressif to migrate the existing blobs we've hacked in with an SoC-specific west extension to the generic solution

Before we can move on to the points described by @mbolivar-nordic above, we need to:

1. Transfer the contents of [the working document](https://docs.google.com/document/d/1heqcv7dzGvM5rA9xpTMW3kyKJLpZsl2Gjsmr5eqBje8/edit#heading=h.a4tmntn7vwmi) to the official Zephyr doc
2. Define the process by which binary blobs are accepted by the TSC. I suggest expanding the [external components](https://docs.zephyrproject.org/latest/contribute/external.html) section.

If you'd like we can collaborate on this @npal-cy. I could work on the docs and you could start drafting the west extension?

gmarull commented on dts/arm/infineon/psoc6/psoc6_02/psoc6_02.124-bga.dtsi:
```diff
@@ -0,0 +1,2051 @@
+				#gpio-cells = <2>;
+			};
+
+			/* audioss_clk_i2s_if */
+			p5_0_audioss0_clk_i2s_if: p5_0_audioss0_clk_i2s_if {
+				DT_CAT1_PINCTRL_INFO(5, 0, HSIOM_SEL_ACT_10, AUDIOSS_CLK_I2S_IF);
+			};
+			p9_0_audioss0_clk_i2s_if: p9_0_audioss0_clk_i2s_if {
+				DT_CAT1_PINCTRL_INFO(9, 0, HSIOM_SEL_ACT_10, AUDIOSS_CLK_I2S_IF);
+			};
+			p11_0_audioss1_clk_i2s_if: p11_0_audioss1_clk_i2s_if {
+				DT_CAT1_PINCTRL_INFO(11, 0, HSIOM_SEL_ACT_10, AUDIOSS_CLK_I2S_IF);
+			};
+			p13_0_audioss1_clk_i2s_if: p13_0_audioss1_clk_i2s_if {
+				DT_CAT1_PINCTRL_INFO(13, 0, HSIOM_SEL_ACT_10, AUDIOSS_CLK_I2S_IF);
+			};
```
why do we have all such pinctrl here? pinctrl pre-definitions are usually package dependent, however, stuff like peripherals is not, this file seems to be mixing things (maybe I'm missing something)

npal-cy replied:
those are pre-defined pin controls nodes. in last updates it will looks like (expect to push at the end of this week):
'''
			/omit-if-no-ref/ p11_1_scb5_i2c_sda: p11_1_scb5_i2c_sda {
				pinmux = <DT_CAT1_PINMUX(11, 1, HSIOM_SEL_ACT_7)>;
			};
'''
And i will remove all nodes which are not currently supported by ifx_cat1 drivers (e.g. audioss_clk_i2s_if)

psoc6_0x_yyy.zzz.dtsi it is actually package related dt, which included from MPN dts.

Infineon CAT1 SoC's devicetree includes a set of pre-defined pin control nodes which can be found via MPN dtsi.
For example board cy8cproto_062_4343w uses CY8C624ABZI_S2D44 part, so in board dts (boards\arm\cy8cproto_062_4343w\cy8cproto_062_4343w.dts) there  is include to MPN dtsi (infineon/psoc6/mpns/CY8C624ABZI_S2D44.dtsi).
 
 Each MPN dtsi includes package dtsi (../psoc6_xx/psoc6_xx.yyy-zzz.dtsi), example for CY8C624ABZI_S2D44 includes "../psoc6_02/psoc6_02.124-bga.dtsi".

gmarull commented on dts/arm/infineon/psoc6/psoc6_02/psoc6_02.124-bga.dtsi:
```diff
@@ -0,0 +1,2051 @@
+
+			/* audioss_pdm_clk */
+			p10_4_audioss0_pdm_clk: p10_4_audioss0_pdm_clk {
+				DT_CAT1_PINCTRL_INFO(10, 4, HSIOM_SEL_ACT_9, AUDIOSS_PDM_CLK);
```
we also agreed on not using macros that hide actual DT code. Macros shall only be used for property content, as it's used e.g. in nrf or stm32/gd32 to encode `pinmux` property.

npal-cy replied:
done

gmarull commented on soc/arm/infineon_cat1/common/pinctrl_soc.h:
```diff
@@ -0,0 +1,129 @@
+#ifndef ZEPHYR_SOC_ARM_INFINEON_CAT1_COMMON_PINCTRL_SOC_H_
+#define ZEPHYR_SOC_ARM_INFINEON_CAT1_COMMON_PINCTRL_SOC_H_
+
+#include <devicetree.h>
+#include <zephyr/types.h>
+#include "dt-bindings/pinctrl/cat1-pinctrl.h"
```
please use `<zephyr/` prefix and avoid `""` for non-local includes

gmarull commented on soc/arm/infineon_cat1/common/pinctrl_soc.h:
```diff
@@ -0,0 +1,129 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
```
this is not needed with SPDX

npal-cy replied:
removed

gmarull commented on soc/arm/infineon_cat1/psoc6/Kconfig.defconfig:
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2021 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_CAT1
+source "soc/arm/infineon_cat1/psoc6/Kconfig.defconfig.soc.*"
+
+config SOC_SERIES
+	default "psoc6"
+
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 100000000
+
+config ROM_START_OFFSET
+	default 0x2000  if SOC_PSOC6_CM0P_SLEEP
```
nit: `0x2000 if`

gmarull commented on soc/arm/infineon_cat1/psoc6/Kconfig.defconfig.soc.psoc6_02:
```diff
@@ -0,0 +1,14 @@
+if SOC_DIE_PSOC6_02
+
+config NUM_IRQS
+	default  32 if CPU_CORTEX_M0PLUS
```
nit: `default 32`

gmarull commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+	bool enable_cts = (data->obj.pin_cts != NC);
+	bool enable_rts = (data->obj.pin_rts != NC);
+
+	if (cfg == NULL) {
+		return -EINVAL;
+	}
```
we typically don't do these checks, it's an API contract to provide a valid config

gmarull commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+	}
+
+	/* Initialize the UART peripheral */
+	result = cyhal_uart_init(&DEV_DATA(dev)->obj,
+				 cat1_pinctrl_get_cyhal_gpio(config->pcfg, UART_TX_PIN),
+				 cat1_pinctrl_get_cyhal_gpio(config->pcfg, UART_RX_PIN),
+				 cat1_pinctrl_get_cyhal_gpio(config->pcfg, UART_CTS_PIN),
+				 cat1_pinctrl_get_cyhal_gpio(config->pcfg, UART_RTS_PIN),
+				 NULL, NULL);
```
you need to use pinctrl, if your HAL takes care of pins you'll need to patch it so that it doesn't. We did the same for nRF (you can check some nRF drivers where you'll see HAL gets a special value to indicate pins are configured somewhere else)

gmarull replied:
This is no longer allowed, you'll need to patch your HAL to not take care of pins (same as Nordic did)

gmarull commented on drivers/gpio/gpio_infineon_cat1.c:
```diff
@@ -0,0 +1,332 @@
+#include "cyhal_hwmgr.h"
+#include "cy_gpio.h"
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(gpio_cat1);
```
```c
#include <logging/log.h>
LOG_MODULE_REGISTER(gpio_cat1, CONFIG_GPIO_LOG_LEVEL);
```

gmarull commented on drivers/gpio/gpio_infineon_cat1.c:
```diff
@@ -0,0 +1,332 @@
+	sys_slist_t callbacks;
+};
+
+#define DEV_CFG(dev) \
+	((const struct gpio_cat1_config *const)(dev)->config)
+
+#define DEV_DATA(dev) \
+	((struct gpio_cat1_data *const)(dev)->data)
```
ditto, please remove

gmarull commented on drivers/gpio/gpio_infineon_cat1.c:
```diff
@@ -0,0 +1,332 @@
+		event = CYHAL_GPIO_IRQ_RISE;
+		break;
+
+	case GPIO_INT_TRIG_BOTH:
+		/* Trigger detection on pin rising or falling edge (GPIO_INT_TRIG_BOTH)
+		 * is not supported. Refer to SWINTEGRATION-696
+		 */
+		/* event = CYHAL_GPIO_IRQ_BOTH; */
+		return -ENOTSUP;
```
you can just remove this case, default will return the same

yurii2001lozyn replied:
Keep this case for clear understanding that GPIO_INT_TRIG_BOTH doesn't support so far.

gmarull commented on drivers/gpio/gpio_infineon_cat1.c:
```diff
@@ -0,0 +1,332 @@
+	cyhal_gpio_callback_data_t *cb_data_ptr = cfg->cb_data_ptr;
+
+	/* Find index of free callback data structure */
+	uint32_t index;
```
nit, please move to the top

gmarull commented on drivers/gpio/gpio_infineon_cat1.c:
```diff
@@ -0,0 +1,332 @@
+			      CONFIG_KERNEL_INIT_PRIORITY_DEVICE,		     \
+			      &gpio_cat1_api);
+
+DT_INST_FOREACH_STATUS_OKAY(GPIO_CAT1_INIT)
```
can you confirm GPIO tests pass for this driver?

npal-cy replied:
we run those GPIO tests:
INFO    - 264/266 cy8cproto_062_4343w       tests/drivers/gpio/gpio_basic_api/drivers.gpio.2pin PASSED (device 76.524s)
INFO    - 265/266 cy8cproto_062_4343w       tests/drivers/gpio/gpio_api_1pin/peripheral.gpio.1pin PASSED (device 74.724s)
[peripheral.gpio.1pin_handler.log](https://github.com/zephyrproject-rtos/zephyr/files/8889072/peripheral.gpio.1pin_handler.log)
[drivers.gpio.2pin_handler.log](https://github.com/zephyrproject-rtos/zephyr/files/8889073/drivers.gpio.2pin_handler.log)

gmarull commented on drivers/pinctrl/pinctrl_infineon_cat1.c:
```diff
@@ -0,0 +1,234 @@
+ * Entries will be NULL if the GPIO port is not enabled.
+ */
+static GPIO_PRT_Type *const gpio_ports[] = {
+	(GPIO_PRT_Type *) GPIO0_REG_ADDR,
```
suggestion here:

you can create a macro like `GPIO_PORT_OR_NULL` like this:

```c
#define GPIO_PORT_OR_NULL(node_id) \
    COND_CODE1(DT_NODE_EXISTS(node_id), ((GPIO_PRT_Type *)DT_REG_ADDR(node_id)), (NULL))

static GPIO_PRT_Type *const gpio_ports[] = {
	GPIO_PORT_OR_NULL(DT_NODELABEL(gpio_port0)),
        ...
};
```

gmarull commented on drivers/pinctrl/pinctrl_infineon_cat1.c:
```diff
@@ -0,0 +1,234 @@
+	int ret;
+	const struct pinctrl_state *state;
+
+	ret = pinctrl_lookup_state(config, PINCTRL_STATE_DEFAULT, &state);
+	if (ret != 0) {
+		return NC;
+	}
```
why does that pinctrl driver run lookups?

gmarull commented on dts/bindings/pinctrl/infineon,mtbhal-pinctrl.yaml:
```diff
@@ -0,0 +1,30 @@
+
+compatible: "infineon,mtbhal-pinctrl"
+
+include: [base.yaml, pincfg-node.yaml]
```
please just allow valid properties, check other bindings

gmarull reviewed (changes requested):
Thanks for the continued work, please check my comments (stopped review just before clock control for now)

npal-cy commented:
@carlescufi , @gmarull , @mbolivar-nordic thanks for comments. I will try address it as soon as possible..

npal-cy commented on dts/arm/infineon/psoc6/mpns/CY8C624ABZI_S2D44.dtsi:
```diff
@@ -0,0 +1,59 @@
+/*
```
Correct. the board should choose/use MPN and doesn't care about common part (psoc6_xx_dtsi) or package part (psoc6_xx.<package_type>_dtsi).

so MPN (manufacture part number) e.g. CY8C624ABZI_S2D44.dtsi includes psoc6_02.dtsi (common part for some die type) and psoc6_02.124-bga.dtsi (package_type)

example of using this from board you can see here: boards/arm/cy8cproto_062_4343w/cy8cproto_062_4343w.dts (refer to line#7)

Board just include MPN which currently soldered on this board. 
To decrease number of files for review the dts and soc has files only for one MPN (CY8C624ABZI_S2D44), and die family (psoc6_02)

Finally, we will provide  dts and soc for all PSoC6 MPNs (in separate PR ), you can look at how it will looks like:
https://github.com/ifyall/zephyr/tree/develop/dts/arm/infineon/psoc6/mpns
https://github.com/ifyall/zephyr/tree/develop/dts/arm/infineon/psoc6

npal-cy commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+		ret = uart_cat1_configure(dev, &config->z_cfg);
+	}
+
+	return ret;
```
we must return error if cyhal_uart_init fails. please update.

nandojve commented on soc/arm/infineon_cat1/psoc6/Kconfig.soc:
```diff
@@ -0,0 +1,117 @@
+	  The Cortex-M0+ application code is placed to internal flash by the Cortex-M4 linker script.
+
+config SOC_PSOC6_CM0P_SLEEP
+	bool "DeepSleep"
```
This is definitely is not the correct approach/name. I mean, those MCUs are asymmetric dual core. There is no problem to build Cortex-M4 with buildin CM0+ but this name is not good and don't tell to anyone the real meaning.

In the current implementation M0+ is always available and M4 can be enabled by user. https://github.com/zephyrproject-rtos/zephyr/blob/b9e833366f1360ecbad8776d6e8587e180688e1b/soc/arm/cypress/Kconfig#L24-L28
This is an small convenience because M0+ boots M4.

I think a similar strategy could be made here where user can decide if he wants a prebuild image integrated or he will build a separeted image to both Cortex-M0+ and M4. This could be a hidden symbol initially to always support the prebuild image (default).

npal-cy replied:
@nandojve, 
per https://github.com/zephyrproject-rtos/zephyr/issues/42883 section "Cortex-M0+ support":
_"In the PSoC 6-based integration, the family's Cortex-M0+ will be controlled by a pre-built image, rather than being open to the user. This aligns with our current PSoC 6 strategy. This pre-built image will be acquired from the Infineon MCU SDK via west extensions the user will invoke when creating a PSoC 6-based application. In future revisions of our Zephyr support, including when adding new device families, we will seek to open the Cortex-M0+ for user development."_

i ok to rename: 
SOC_PSOC6_CM0P_SLEEP -> SOC_PSOC6_CM0P_IMAGE_SLEEP
SOC_PSOC6_CM0P_****** -> SOC_PSOC6_CM0P_IMAGE_****

@ifyall, please provide your thoughts.

ifyall replied:
A rename to make the intent clear makes sense to me.
Regarding the prompt, perhaps we can change it to something like this:

`choice
	prompt "PSoC™ 6 Cortex M0+ prebuilt images"
	help
	  Choose the prebuilt application image to be executed on the Cortex-M0+ core of the PSoC™ 6 dual - core MCU. The image is responsible for booting the Cortex-M4 on the device.`

npal-cy replied:
done

nandojve commented on soc/arm/infineon_cat1/psoc6/noinit.ld:
```diff
@@ -0,0 +1,7 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Extracted from:
```
?

nandojve commented on soc/arm/infineon_cat1/psoc6/ram_cm0image.ld:
```diff
@@ -0,0 +1,16 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Extracted from:
+ * ext/hal/cypress/.../devices/psoc6/linker/gcc/cy8c6xx6_cm0plus.ld
+ */
+
+/* Cortex-M0+ application ram image area */
+/*
+ * Copyright (c) 2021 Cypress Semiconductor Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Is it possible unify this ?

nandojve commented on soc/arm/infineon_cat1/psoc6/rom_cm0image.ld:
```diff
@@ -0,0 +1,11 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
```
Not sure if copyright is required to avoid twister complain.

nandojve commented on soc/arm/infineon_cat1/psoc6/soc.c:
```diff
@@ -0,0 +1,77 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
```
not necessary with SPDX

nandojve commented on soc/arm/infineon_cat1/psoc6/soc.h:
```diff
@@ -0,0 +1,32 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
```
ditto

nandojve commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+#define DEV_CFG(dev) \
+	((struct uart_cat1_config *const)(dev)->config)
+
+/* Internal function to get cyhal_gpio_t from pinctrl array by defined index. */
+extern cyhal_gpio_t cat1_pinctrl_get_cyhal_gpio(const struct pinctrl_dev_config *config,
+						uint8_t signal_name);
```
Why is this not handled at pinctrl driver directly?

nandojve commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+}
+
+
+/* UART CONSOLE AND CONFIGURATION API */
```
Drop all useless comments.

nandojve commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+	return ((rec == CY_SCB_UART_RX_NO_DATA) ? -1 : 0);
+}
+
+static void uart_cat1_poll_out(const struct device *dev, unsigned char c)
+{
+	(void) cyhal_uart_putc(&DEV_DATA(dev)->obj, (uint32_t)c);
+}
```
What happen if buffer is full? Is this blocking?

npal-cy replied:
https://github.com/zephyrproject-rtos/zephyr/pull/44211#issuecomment-1202399515

nandojve commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+		result = cyhal_uart_enable_flow_control(&data->obj, enable_cts,
+							enable_rts);
+	}
+	return (result == CY_RSLT_SUCCESS) ? 0 : -1;
```
Please, use errno constants to return errors.

npal-cy replied:
updated

nandojve commented on dts/bindings/serial/infineon,mtbhal-uart.yaml:
```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2022, Cypress
```
This is a copy with few changes from https://github.com/zephyrproject-rtos/zephyr/blob/b9e833366f1360ecbad8776d6e8587e180688e1b/dts/bindings/serial/cypress%2Cpsoc6-uart.yaml#L1-L34

You MUST keep copyrights

```diff
+# Copyright (c) 2018, Cypress
+# Copyright (c) 2020, ATL Electronics
# Copyright (c) 2022, Cypress
```

npal-cy replied:
done

nandojve commented on dts/bindings/serial/infineon,mtbhal-uart.yaml:
```diff
@@ -0,0 +1,76 @@
+    interrupts:
+      required: true
+
+    peripheral-id:
+      type: int
+      description: peripheral ID
+      required: true
+
```
Why you need this?

nandojve commented on dts/bindings/serial/infineon,mtbhal-uart.yaml:
```diff
@@ -0,0 +1,76 @@
+      description: peripheral ID
+      required: true
+
+    pinctrl-0:
+      description: |
+        PORT pin configuration for TX, RX, RTS, CTS signals.
+        We expect that the phandles will reference pinctrl nodes These
+        nodes will have a nodelabel that matches the Infineon SoC Pinctrl
+        defines and be of the form p<port>_<pin><peripheral inst>_<signal>.
+
+        Examples
+           use TX, RX
+           pinctrl-0 = <&p5_1_scb5_uart_tx &p5_0_scb5_uart_rx>;
+
+           use RX only
+           pinctrl-0 = <&p5_0_scb5_uart_rx>;
+
+           use TX, RX, RTS, CTS
+           pinctrl-0 = <&p3_1_scb2_uart_tx &p3_0_scb2_uart_rx &p3_2_scb2_uart_rts &p3_3_scb2_uart_cts>;
+      required: true
+
+    pinctrl-names:
+      required: true
+
```
@gmarull Is this following pinctrl states ? Shouldn't user just include `pinctrl-device.yaml` ?

nandojve reviewed (changes requested):
Hi @npal-cy ,

I noted that commits are not in order. I mean, pinctrl and clock are pre-requirements for other drivers. Once we will merge all together the order should be respected.

I stop review before gpio because I think there are too many requests in place.

ifyall commented on CODEOWNERS:
```diff
@@ -38,9 +38,10 @@
 /soc/arm/atmel_sam/same70/                @nandojve
 /soc/arm/atmel_sam/samv71/                @nandojve
-/soc/arm/cypress/                         @nandojve
+/soc/arm/cypress/                         @nandojve @ifyall
```
Nazar, please add yourself as well.

npal-cy replied:
added

npal-cy commented on soc/arm/infineon_cat1/psoc6/CMakeLists.psoc6_02.txt:
```diff
@@ -0,0 +1,38 @@
+zephyr_include_directories(.)
+
+# add MPN defines
+zephyr_compile_definitions_ifdef(CONFIG_SOC_CYB0644ABZI_S2D44 CYB0644ABZI_S2D44)
```
i will update per @nandojve suggestion.

npal-cy replied:
updated

dcpleung commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+#include <device.h>
+#include <drivers/uart.h>
+#include <drivers/pinctrl.h>
+#include "cyhal_uart.h"
```
Please use `<...>` as this is not a local include.

npal-cy replied:
updated

dcpleung commented on dts/bindings/serial/infineon,mtbhal-uart.yaml:
```diff
@@ -0,0 +1,76 @@
+        The number of stop bits, refer to uart_config_stop_bits
+      default: 1
+
+    data_bits:
+      required: false
+      type: int
+      description: |
+        The number of data bits:
+        for 8 use 3 UART_CFG_DATA_BITS_8,
+        for 9 use 4 UART_CFG_DATA_BITS_9,
+        refer to uart_config_data_bits.
+      default: 3
```
Might be better to use `enum` type.

dcpleung commented on drivers/serial/uart_infineon_cat1.c:
```diff
@@ -0,0 +1,506 @@
+/* Device config structure */
+struct uart_cat1_config {
+	const struct pinctrl_dev_config *pcfg;
+	struct uart_config z_cfg;
```
Please don't use `z_` prefix in driver files.

npal-cy replied:
updated

npal-cy commented:
@nandojve, 
I do not know why i can't add answer under your comments. I got error "Could not resolve to a node with the global id of 'MDE3OlB1bGxSZXF1ZXN0UmV2aWV3'."...

 regarding to https://github.com/zephyrproject-rtos/zephyr/pull/44211#discussion_r895001593

Yes, this is a blocking call which waits till the character is sent to UART FIFO. If the buffer is full it will for free place in FIFO.

yurii2001lozyn commented:
@nandojve
regarding to https://github.com/zephyrproject-rtos/zephyr/pull/44211#discussion_r895002732
Peripheral-id used to select proper uart block from dts

yurii2001lozyn commented:
@carlescufi
regarding to https://github.com/zephyrproject-rtos/zephyr/pull/44211#discussion_r893606892
Per DS, it is correct to have 13 i2c instances, 9 spi instances and 12 UARTs, as in current dts psoc6_02.dtsi file.
https://www.infineon.com/dgdl/Infineon-PSOC_6_MCU_CY8C62X8_CY8C62XA-DataSheet-v17_00-EN.pdf?fileId=8ac78c8c7d0d8da4017d0ee7d03a70b1

nandojve commented:
> @nandojve regarding to [#44211 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/44211#discussion_r895002732) Peripheral-id used to select proper uart block from dts

For this  function you must use clocks property.

npal-cy commented on dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi:
```diff
@@ -0,0 +1,994 @@
+			label = "adc_0";
+		};
+
+		i2c0: i2c@40600000 {
```
double check  13 i2c instances, 9 spi instances and 12 UARTs, as in current dts psoc6_02.dtsi file is correct.

npal-cy commented on soc/arm/infineon_cat1/psoc6/Kconfig.series:
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2021 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# Cypress PSoC™ 6 MCU lines
+config SOC_SERIES_PSOC_60
+	bool "Infineon PSoC™ 60 series MCU(Value Line)"
```
i do not see the problem here:
![image](https://user-images.githubusercontent.com/46566584/183926819-841b7b14-f0ea-441c-822a-7644405af290.png)

![image](https://user-images.githubusercontent.com/46566584/183927151-a4f77167-5631-453e-947e-f9e4a5b8c9f0.png)

npal-cy commented on soc/arm/infineon_cat1/Kconfig:
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+
+config SOC_FAMILY_CAT1
```
renamed

galak commented on dts/bindings/serial/infineon,mtbhal-uart.yaml:
```diff
@@ -0,0 +1,76 @@
+      required: true
+
+    current-speed:
+      default: 115200
```
`current-speed` should not have a default in binding

galak commented on dts/bindings/serial/infineon,mtbhal-uart.yaml:
```diff
@@ -0,0 +1,76 @@
+      default: 115200
+
+    stop-bits:
+      required: false
```
remove `defaults` from all the properties in here.

galak commented on dts/bindings/gpio/infineon,mtbhal-gpio.yaml:
```diff
@@ -0,0 +1,28 @@
+
+    interrupts:
+      required: true
+
```
remove `label` property

galak commented on dts/arm/infineon/psoc6/mpns/CY8C624ABZI_S2D44.dtsi:
```diff
@@ -0,0 +1,59 @@
+		reg = < 0x40240000 0x10000 >;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		label = "CYPRESS_FLASH_CONTROLLER";
```
remove label property in .dtsi

npal-cy replied:
please confirm, the label must removed for all nodes, am i right?

galak commented on drivers/serial/Kconfig.ifx_cat1:
```diff
@@ -0,0 +1,17 @@
+# an affiliate of Cypress Semiconductor Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_INFINEON_CAT1_UART := infineon,mtbhal-uart
```
Use new style:

remove DT_COMPAT_INFINEON_CAT1_UART

galak commented on drivers/serial/Kconfig.ifx_cat1:
```diff
@@ -0,0 +1,17 @@
+DT_COMPAT_INFINEON_CAT1_UART := infineon,mtbhal-uart
+
+config UART_INFINEON_CAT1
+	bool "Infineon CAT1 UART driver"
```
change this to:

```
default y
depends on DT_HAS_INFINEON_MTBHAL_UART
```

galak commented on drivers/serial/Kconfig.ifx_cat1:
```diff
@@ -0,0 +1,17 @@
+config UART_INFINEON_CAT1
+	bool "Infineon CAT1 UART driver"
+		default $(dt_compat_enabled,$(DT_COMPAT_INFINEON_CAT1_UART))
+	depends on SOC_FAMILY_INFINEON_CAT1
```
remove depends on SOC_FAMILY_INFINEON_CAT1, its implied by `depends on DT_HAS_INFINEON_MTBHAL_UART

galak reviewed (changes requested):
dts & Kconfig changes.

Also, the new bindings need different compatibles to match the HW not HAL.

npal-cy commented:
> For this function you must use clocks property.
in dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi, we use peripheral-id to define block number
```
		uart0: uart@40600000 {
			compatible = "infineon,mtbhal-uart";
			reg = <0x40600000 0x10000>;
			interrupts = <39 6>;
			peripheral-id = <0>;
			status = "disabled";
			label = "uart_0";
		};
		uart1: uart@40610000 {
			compatible = "infineon,mtbhal-uart";
			reg = <0x40610000 0x10000>;
			interrupts = <40 6>;
			peripheral-id = <1>;
			status = "disabled";
			label = "uart_1";
		};
		uart2: uart@40620000 {
			compatible = "infineon,mtbhal-uart";
			reg = <0x40620000 0x10000>;
			interrupts = <41 6>;
			peripheral-id = <2>;
			status = "disabled";
			label = "uart_2";
		};
...
```
driver drivers/serial/uart_ifx_cat1.c use this block_num during initialization in ifx_cat1_uart_init function.
info stores in config structure from INFINEON_CAT1_UART_INIT macro:
```
#define INFINEON_CAT1_UART_INIT(n)
...
     hw_resource = {					     \
	.type = CYHAL_RSC_SCB,			     \
	.block_num = DT_INST_PROP(n, peripheral_id), \
	}
```

npal-cy commented:
> Also, the new bindings need different compatibles to match the HW not HAL.

hi @galak, 
we include 'mtbhal' in binding name (e.g infineon,mtbhal-uart.yaml), it means that all our our socs, which supports be our mtbhal will use the same bindings/drivers. For now, in this PR we integrating PSOC6, but we expect reuse same bindings/drivers for our new family of socs.

Do you see problem here?

galak commented:
> > 
> 
> hi @galak, we include 'mtbhal' in binding name (e.g infineon,mtbhal-uart.yaml), it means that all our socs, which supports by our mtbhal will use the same bindings/drivers. For now, in this PR we integrating PSOC6, but we expect reuse same bindings/drivers for our new family of socs.
> 
> Do you see problem here?

The devicetree compatibles should describe the hardware as if no HAL ever existed.  Its fine for the driver to utilize the HAL and support multiple different compatibles, but the devicetree itself should use compatibles that are descriptive of the hardware.

npal-cy commented:
> > > 
> > 
> > 
> > hi @galak, we include 'mtbhal' in binding name (e.g infineon,mtbhal-uart.yaml), it means that all our socs, which supports by our mtbhal will use the same bindings/drivers. For now, in this PR we integrating PSOC6, but we expect reuse same bindings/drivers for our new family of socs.
> > Do you see problem here?
> 
> The devicetree compatibles should describe the hardware as if no HAL ever existed. Its fine for the driver to utilize the HAL and support multiple different compatibles, but the devicetree itself should use compatibles that are descriptive of the hardware.

ok, we will rename form “Infineon, mtbhal-xxxx.." to “Infineon, cat1-xxxx.."

jsiverskog commented:
@npal-cy : any plan to finish this and getting it merged?

ifyall commented:
> 

@jsiverskog, we are working on the required binary acquisition that is blocking this PR. There is quite a bit of review/comments here and so ideally, we would keep the PR open. Our goal is to have this PR ready by end of the month.

Thanks,

Ian

ifyall commented:
Infineon will be updating this PR during the next two weeks and will ask for re-review when ready.

npal-cy committed:
modules: hal_infineon: Added blob support for CM0+ images

Initial blob support for PSoC6 CM0+ images

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy commented:
Hi @carlescufi 
We added blobs support in https://github.com/zephyrproject-rtos/zephyr/pull/44211/commits/64c2ed0ee5f31e31413b7bde7ecf90a7b814d7d6. It has dependency on hal_infineon repo updates.

All Twister build test are failed with cy8cproto_062_4343w board.
https://github.com/zephyrproject-rtos/zephyr/actions/runs/4054950354/jobs/6977556794
ninja: error: 
> '/__w/zephyr/modules/hal/infineon/zephyr/blobs/img/cat1cm0p/COMPONENT_CM0P_SLEEP/psoc6_02_cm0p_sleep.bin', needed by 'zephyr/include/generated/psoc6_cm0p_sleep.bin.inc', missing and no known rule to make it

Do we have any defined way to enable _west blobs fetch_ for this board?

Regards,
Nazar

mbolivar-nordic commented:
Hi @npal-cy 

> Do we have any defined way to enable _west blobs fetch_ for this board for Twister test?

I'm afraid not. The TSC voted explicitly against enabling CI testing for components that require blobs.

npal-cy commented:
hi @mbolivar-nordic,
Do you have any suggestion how we can enable Twister for cy8cproto_062_4343w board? as i am understanding with current status of CI this MR is blocked.

Thanks,
Nazar

mbolivar-nordic commented:
> Do you have any suggestion how we can enable Twister for cy8cproto_062_4343w board? as i am understanding with current status of CI this MR is blocked.

From memory, the short version of the deal that the project made with
contributors who want to submit things that require binary blobs
upstream is:

- the contributions won't get any CI testing
- it's the contributor's responsibility to test and ensure things work

I haven't looked at the PR's current status, but I would guess there
should be some way to limit twister from exercising your files to abide
by this deal. You'd have to take it up with the twister maintainers if
it's not clear from the twister documentation what that way is.

npal-cy commented:
hi @mbolivar-nordic , thanks for your suggestion.

From Twister documentation i see only one option to use 'platform_exclude', but it is not suitable for us, because:
- we want that our platform build (to catch any regression)
- to use 'platform_exclude' need to update all projects in Zephyr tree. 

One option which i see here:  I'll update my blobs integration modules\hal_infineon\cat1cm0p\CMakeLists.txt to include CM0+ image sources (psoc6_cm0p_sleep.c) only if we found binary file and could generate binaries file include (by using generate_inc_file_for_target). If we can't find binary file Cmake generates warning "Can't find CM0+ binaries, please execute: \'west blobs fetch hal_infineon\'

It gives us possibility to have successful build in Twister even if west blobs fetch was not executed. From user protective:
- In  board documentation we have instruction about Fetch Binary Blobs.
- in build log cmake will generate Warning "Can't find CM0+ binaries, please execute: \'west blobs fetch hal_infineon\'"

npal-cy commented:
Hi @carlescufi, @gmarul, @nandojve, @mbolivar-nordic, @galak,
All comments are addressed, CI pass. Please do final review.

Thanks and Regards,
Nazar

nandojve commented:
Hi @npal-cy ,

I would like to ask you to remove my reference on CODEOWNERS and collaboration on MAINTAINERS file related to CY.
This means that after this PR got merged I should not be included anymore to participate on this architecture.

BR,
Gerson

npal-cy commented:
> Hi @npal-cy ,
> 
> I would like to ask you to remove my reference on CODEOWNERS and collaboration on MAINTAINERS file related to CY. This means that after this PR got merged I should not be included anymore to participate on this architecture.
> 
> BR, Gerson

Hi Gerson,

Done. Please double check https://github.com/zephyrproject-rtos/zephyr/pull/44211/commits/ed2ec5b1ea448cc5369d7c5388f12c0d6fbf6c02.

Regards,
Nazar

galak reviewed (commented):
I would suggest pulling out the CODEOWNER/MAINTAINERS.yml change into a separate PR.

npal-cy committed:
MAINTAINERS.yml: Update maintainers for Cypress sources

- Remove Gerson Fernando Budke (@nandojve) from maintainers
of Cypress sources.

- Add Nazar Palamar (@npal-cy) as collaborators for
Infineon/Cypress sources.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

nandojve commented:
> > Hi @npal-cy ,
> > I would like to ask you to remove my reference on CODEOWNERS and collaboration on MAINTAINERS file related to CY. This means that after this PR got merged I should not be included anymore to participate on this architecture.
> > BR, Gerson
> 
> Hi Gerson,
> 
> Done. Please double check [ed2ec5b](https://github.com/zephyrproject-rtos/zephyr/commit/ed2ec5b1ea448cc5369d7c5388f12c0d6fbf6c02).
> 
> Regards, Nazar

You should a PR then I can review.

npal-cy commented:
Hi Gerson,

The changes was done in those two commits:
[CODEOWNERS: Update codeowners for Infineon/Cypress sources](https://github.com/zephyrproject-rtos/zephyr/pull/44211/commits/dcb150314dce83192ff04dca8dc827f89a9713ec) 

[MAINTAINERS.yml: Update maintainers for Cypress sources](https://github.com/zephyrproject-rtos/zephyr/pull/44211/commits/f9b6db56e07298ae1956d1f1760e0e09896aa3ca) 

Thanks,
Nazar

nandojve reviewed (dismissed):
+1 for CODEOWNERS and MAINTAINERS changes,

npal-cy commented:
> I would suggest pulling out the CODEOWNER/MAINTAINERS.yml change into a separate PR.
sorry i missed "PR", i pulled out to different commits.  Let me change this.

npal-cy committed:
west.yml: Update hal_infineon revision

Get blob support for cm0+ in hal_infineon

Updated revision of hal_infineon to:
8485083ad91d3e2cc5d706da3464716718a6a42e

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

carlescufi commented:
@gmarull @mbolivar-nordic @galak please take another look

mniestroj commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,195 @@
+Errata
+====================
+
++------------------------------------------------+----------------------------------------+
+| Problem                                        | Solution                               |
++------------------------------------------------+----------------------------------------+
```
```suggestion
+------------------------------------------------+----------------------------------------+
| Problem                                        | Solution                               |
+================================================+========================================+
```

npal-cy replied:
updated

mniestroj commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,195 @@
+.. _cy8cproto_062_4343w:
+
+CY8CPROTO-062-4343W
+#########################
```
```suggestion
CY8CPROTO-062-4343W
###################
```

mniestroj commented on west.yml:
```diff
@@ -73,7 +73,7 @@ manifest:
         - hal
     - name: hal_infineon
-      revision: 5b0c7700a7be049e02835dfa0950d378aac39e67
+      revision: pull/6/head
```
Should this be a sha1 instead?

npal-cy replied:
was updated in https://github.com/zephyrproject-rtos/zephyr/pull/44211/commits/6ba2ac774e528c672ba4ee3126090efbeb7c3b1c

mniestroj replied:
I see that. But it will make `git bisect` break in the future with this dangling reference (`pull/6/head`).

npal-cy replied:
rebased.

galak commented on MAINTAINERS.yml:
```diff
@@ -2343,7 +2343,6 @@ West:
   maintainers:
     - ifyall
   collaborators:
-    - nandojve
```
any reason `@npal-cy` isn't listed as collaborator?  Feels like this file needs updating for all this.

npal-cy replied:
added.

galak commented on boards/arm/cy8cproto_062_4343w/cy8cproto_062_4343w.dts:
```diff
@@ -0,0 +1,104 @@
+
+	/* Configure pin control bias mode for i2c3 pins */
+	p6_0_scb3_i2c_scl {
+		drive-open-drain; input-enable;
```
Please fix this

npal-cy replied:
updated

galak commented on boards/arm/cy8cproto_062_4343w/cy8cproto_062_4343w.dts:
```diff
@@ -0,0 +1,104 @@
+	};
+
+	p6_1_scb3_i2c_sda {
+		drive-open-drain; input-enable;
```
ditto

galak commented on boards/arm/cy8cproto_062_4343w/cy8cproto_062_4343w.yaml:
```diff
@@ -0,0 +1,18 @@
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
```
remove xtools, we've deprecated the support

npal-cy replied:
done

galak commented on dts/bindings/serial/infineon,cat1-uart.yaml:
```diff
@@ -0,0 +1,72 @@
+      - "1"
+      - "1.5"
+      - "2"
+    default: "1"
```
remove default here.

npal-cy replied:
removed. Updated uart_ifx_cat1.c driver to handle default values for stop_bits/data_bits:
```
.#define INFINEON_CAT1_UART_INIT(n)							     \
...
	static struct ifx_cat1_uart_config ifx_cat1_uart##n##_cfg = {			     \
	   .dt_cfg.stop_bits = DT_INST_ENUM_IDX_OR(n, stop_bits, UART_CFG_STOP_BITS_1), \
	   .dt_cfg.data_bits = DT_INST_ENUM_IDX_OR(n, data_bits, UART_CFG_DATA_BITS_8), \
....
```

galak commented on dts/bindings/serial/infineon,cat1-uart.yaml:
```diff
@@ -0,0 +1,72 @@
+      - 7
+      - 8
+      - 9
+    default: 8
```
remove default here.

npal-cy replied:
done

galak reviewed (dismissed):
Various things need updating

npal-cy commented on drivers/gpio/Kconfig.ifx_cat1:
```diff
@@ -0,0 +1,12 @@
+	default y
+	depends on DT_HAS_INFINEON_CAT1_GPIO_ENABLED
+	help
+		Enable support for Infineon CAT1 GPIO controllers.
```
updated

npal-cy commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,195 @@
+
+Kit Contents:
+==================
+- PSoC 6 Wi-Fi BT Prototyping Board
```
done

npal-cy commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,195 @@
+
+Kit Features:
+==================
+- Support of up to 2MB Flash and 1MB SRAM
```
done

npal-cy commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,195 @@
+
+Fetch Binary Blobs
+*******************************************
+cy8cproto_062_4343w board requires fetch binary files
```
done

npal-cy commented on dts/bindings/serial/infineon,cat1-uart.yaml:
```diff
@@ -0,0 +1,72 @@
+    default: 8
+
+  parity:
+    default: "none"
```
done

npal-cy commented:
Hi @mniestroj and @galak,

Thanks for new portion of comments. I addressed them please review.

Regards,
Nazar

gmarull commented on drivers/serial/uart_ifx_cat1.c:
```diff
@@ -0,0 +1,506 @@
+	struct uart_config cfg;
+	cyhal_clock_t clock;
+
+	#if CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_user_data_t irq_cb;           /* Interrupt Callback */
+	void *irq_cb_data;                              /* Interrupt Callback Arg */
+	#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
```
`#if` usually starts at col 0 (or check with clang-format results)

npal-cy replied:
Updated. I am using uncrustify, but now i see Zephyr does not provide uncrustify configuration... I will look and integrate in our CI clang-format.

gmarull commented on drivers/serial/uart_ifx_cat1.c:
```diff
@@ -0,0 +1,506 @@
+	int ret;
+
+	/* Default Counter configuration structure */
+	static const cy_stc_scb_uart_config_t _cyhal_uart_default_config = {
```
static on a multi-instance driver?

npal-cy replied:
this is `static const` to be sure that this structure locates in RO section..

mniestroj replied:
Personally I like to have `static` (and `static const`) outside of function body. It is easier to locate those differentiate from stack defined variables. It also clearly makes such structure a "configuration struct" and outside of function body, which is usually a "logic implementation".

Not sure whether we have guidelines in Zephyr that say the same.

gmarull commented on dts/bindings/serial/infineon,cat1-uart.yaml:
```diff
@@ -0,0 +1,67 @@
+  interrupts:
+    required: true
+
+  peripheral-id:
+    type: int
+    description: peripheral ID
+    required: true
```
where/why is this used?

npal-cy replied:
we use it to get uart/scb block instance:
```
#define INFINEON_CAT1_UART_INIT(n)							     \
...
		.hw_resource = {							     \
				.type = CYHAL_RSC_SCB,					     \
				.block_num = DT_INST_PROP(n, peripheral_id),		     \
			},	
...
```
this uses for Uart initialization `cyhal_uart_init_cfg()`  in `ifx_cat1_uart_init`  function.

gmarull commented on drivers/gpio/gpio_ifx_cat1.c:
```diff
@@ -0,0 +1,317 @@
+	const struct device *dev = port_dev_obj[port_num];
+
+	/* Goes through and fires callback from a callback list */
+	gpio_fire_callbacks(&((struct gpio_cat1_data *const)(dev)->data)->callbacks,
```
you're dereferencing dev without checking if NULL (it can be according to GET_DEV_OBJ).

gmarull commented on boards/arm/cy8cproto_062_4343w/cy8cproto_062_4343w.dts:
```diff
@@ -0,0 +1,106 @@
+	status = "okay";
+	current-speed = <115200>;
+
+	/* UART pins */
```
redundant comment/s

npal-cy replied:
done

gmarull commented on boards/arm/cy8cproto_062_4343w/cy8cproto_062_4343w.dts:
```diff
@@ -0,0 +1,106 @@
+	clock-div = <1>;
+};
+
+&pinctrl {
+
+	p3_3_scb2_uart_cts {
+		input-enable;
+	};
+
+	/* Configure pin control bias mode for uart5 pins */
+	p5_1_scb5_uart_tx {
+		drive-push-pull;
+	};
+
+	p5_0_scb5_uart_rx {
+		input-enable;
+	};
+
+	/* Configure pin control bias mode for i2c3 pins */
+	p6_0_scb3_i2c_scl {
+		drive-open-drain;
+		input-enable;
+	};
+
+	p6_1_scb3_i2c_sda {
+		drive-open-drain;
+		input-enable;
+	};
+};
```
according to the guidelines, this needs to be put under `board-pinctrl.dtsi`. Also, you can directly extend pin nodes, ie use `&p3_3_scb2_uart_cts`.

npal-cy replied:
done

gmarull commented on boards/arm/cy8cproto_062_4343w/cy8cproto_062_4343w_defconfig:
```diff
@@ -0,0 +1,31 @@
+# Enable clock controller
+CONFIG_CLOCK_CONTROL=y
+
+# Enable assert
+CONFIG_ASSERT=y
```
why is this default?

npal-cy replied:
removed

gmarull commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,199 @@
+In addition, support for Digilent's Pmod interface is also provided with this kit.
+
+.. image:: img/board.jpg
+     :width: 887px
```
don't fix width, just provide a properly sized image

npal-cy replied:
updated

gmarull commented on boards/arm/cy8cproto_062_4343w/doc/index.rst:
```diff
@@ -0,0 +1,199 @@
+
+The following steps shows how to fetch Binary Blobs:
+
+1. Get the Zephyr source code
+
+.. code-block:: console
+
+    west init
+
+2. Update Zephyr modules
+
+.. code-block:: console
+
+   west update
+
+3. Update Zephyr modules and export the environment
+
+.. code-block:: console
+
+   west update
+   west zephyr-export
```
these steps are redundant

npal-cy replied:
removed

gmarull commented on modules/hal_infineon/mtb-hal-cat1/CMakeLists.txt:
```diff
@@ -0,0 +1,97 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set(hal_dir               ${ZEPHYR_HAL_INFINEON_MODULE_DIR}/mtb-hal-cat1)
+set(hal_psoc6_dir         ${hal_dir}/COMPONENT_PSOC6HAL)
+set(hal_psoc6_cat1a_dir   ${hal_dir}/COMPONENT_PSOC6HAL/COMPONENT_CAT1A)
```
to preserve bisectability, HAL module needs to come before drivers or any other code.

npal-cy replied:
rebased

gmarull commented on dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi:
```diff
@@ -0,0 +1,889 @@
+			peripheral-id = <0>;
+			status = "disabled";
+		};
+		i2c1: i2c@40610000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40610000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <40 6>;
+			peripheral-id = <1>;
+			status = "disabled";
+		};
+		i2c2: i2c@40620000 {
+			compatible = "infineon,cat1-i2c";
+			reg = <0x40620000 0x10000>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <41 6>;
+			peripheral-id = <2>;
```
All these `peripheral-id` props don't look like hardware description, but _HALisms_ in DT. Other platforms with the same problem, e.g. Nordic, have had to find workarounds since this is not allowed.  cc: @galak @mbolivar-nordic

npal-cy replied:
Updates:
- removed peripheral-id
- use reg_addr instead peripheral-id...

gmarull commented on dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi:
```diff
@@ -0,0 +1,889 @@
+			status = "disabled";
+		};
+		i2c1: i2c@40610000 {
+			compatible = "infineon,cat1-i2c";
```
note: where are all the compatibles for the nodes introduced in this commit?

npal-cy replied:
another drivers (counter, i2c, pwm) will be added in separate MRs after this one.

gmarull replied:
so all these nodes should be added later, when driver/bindings are added.

npal-cy replied:
removed node without drivers support

npal-cy commented on include/zephyr/dt-bindings/pinctrl/ifx_cat1-pinctrl.h:
```diff
@@ -0,0 +1,99 @@
+#define P19 CYHAL_PORT_19
+#define P20 CYHAL_PORT_20
+
+/* Returns CYHAL GPIO from Board device tree GPIO configuration */
+#define DT_GET_CYHAL_GPIO_FROM_DT_GPIOS(node, gpios_prop)				 \
```
no. It can be used inside another drivers, to get cyhal_gpio objects from DT gpios node..

gmarull replied:
it should likely be moved to pinctrl_soc.h then

npal-cy committed:
drivers: clock_control: Add Infineon CAT1 clock control driver

Add initial version of Infineon CAT1 clock control driver.
- supports clock initialization based on board DT configuration.

Added initial version of system_clocks.dtsi for Infineon PSoC 6 SOC.
Includes: clk_imo, path_mux0..4, fll0, pll0, clk_hf0..4, clk_fast,
clk_slow and clk_peri.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy committed:
drivers: pinctrl: Add Infineon CAT1 Pin controller driver

Added initial version of Infineon CAT1 Pin controller driver.
Added initial version of binding file for Infineon CAT1 Pinctrl driver.
Added initial version of dt header for Infineon CAT1 pinctrl driver.

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy commented on dts/arm/infineon/psoc6/system_clocks.dtsi:
```diff
@@ -0,0 +1,146 @@
+		/* path mux1 */
+		path_mux1: path_mux1 {
+			#clock-cells = <0>;
+			compatible = "fixed-factor-clock";
+		clocks = <&clk_imo>;
+				status = "okay";
```
updated

npal-cy commented on soc/arm/infineon_cat1/common/pinctrl_soc.h:
```diff
@@ -0,0 +1,127 @@
+#ifndef ZEPHYR_SOC_ARM_INFINEON_CAT1_COMMON_PINCTRL_SOC_H_
+#define ZEPHYR_SOC_ARM_INFINEON_CAT1_COMMON_PINCTRL_SOC_H_
+
+#include <zephyr/devicetree.h>
+#include <zephyr/types.h>
```
updated

npal-cy commented on soc/arm/infineon_cat1/common/pinctrl_soc.h:
```diff
@@ -0,0 +1,127 @@
+ * @brief Utility macro to initialize pincfg field in #pinctrl_pin_t.
+ * @param node_id Node identifier.
+ */
+#define Z_PINCTRL_CAT1_PINCFG_INIT(node_id) (					   \
+		(DT_PROP(node_id, bias_pull_up) << SOC_GPIO_PULLUP_POS) |	   \
+		(DT_PROP(node_id, bias_pull_down) << SOC_GPIO_PULLDOWN_POS) |	   \
+		(DT_PROP(node_id, drive_open_drain) << SOC_GPIO_OPENDRAIN_POS) |   \
+		(DT_PROP(node_id, drive_open_source) << SOC_GPIO_OPENSOURCE_POS) | \
+		(DT_PROP(node_id, drive_push_pull) << SOC_GPIO_PUSHPULL_POS) |	   \
+		(DT_PROP(node_id, input_enable) << SOC_GPIO_INPUTENABLE_POS))
+
```
rebased

npal-cy commented on soc/arm/infineon_cat1/psoc6/soc.c:
```diff
@@ -0,0 +1,66 @@
+	 * must be enabled). Otherwise it is ignored.
+	 */
+
+#if (CY_CPU_CORTEX_M0P)
+	#error Cy_SysInt_Init does not support CM0p core.
+#endif /* (CY_CPU_CORTEX_M0P) */
```
we should not reach this. we will update this cade when open CM0+ core for user.

npal-cy commented on soc/arm/infineon_cat1/psoc6/soc.c:
```diff
@@ -0,0 +1,66 @@
+	}
+#endif /* defined(CONFIG_DYNAMIC_INTERRUPTS) && defined(CONFIG_GEN_ISR_TABLES) */
+
+	return(status);
```
done

npal-cy commented on dts/bindings/pinctrl/infineon,cat1-pinctrl.yaml:
```diff
@@ -0,0 +1,120 @@
+              };
+          };
+
+  NOTE2 Pre-defined pin control nodes do not have bias pin configuration.
+    The bias configuration can be updated in board.tds
```
updated

npal-cy commented on drivers/serial/uart_ifx_cat1.c:
```diff
@@ -0,0 +1,506 @@
+#include <cyhal_utils_psoc.h>
+#include <cyhal_scb_common.h>
+
+/* Default UART interrupt priority */
+#if !defined(IFX_CAT1_UART_INTERRUPT_PRIORITY)
+	#define IFX_CAT1_UART_INTERRUPT_PRIORITY      (5u)
+#endif
```
re-worked

mniestroj commented on drivers/gpio/gpio_ifx_cat1.c:
```diff
@@ -0,0 +1,318 @@
+	uint32_t pin_num = CYHAL_GET_PIN((uint32_t) callback_arg);
+
+	static const struct device *const port_dev_obj[IOSS_GPIO_GPIO_PORT_NR] = {
+		GET_DEV_OBJ(gpio_prt0),  GET_DEV_OBJ(gpio_prt1),  GET_DEV_OBJ(gpio_prt2),
+		GET_DEV_OBJ(gpio_prt3),  GET_DEV_OBJ(gpio_prt4),  GET_DEV_OBJ(gpio_prt5),
+		GET_DEV_OBJ(gpio_prt6),  GET_DEV_OBJ(gpio_prt7),  GET_DEV_OBJ(gpio_prt8),
+		GET_DEV_OBJ(gpio_prt9),  GET_DEV_OBJ(gpio_prt10), GET_DEV_OBJ(gpio_prt11),
+		GET_DEV_OBJ(gpio_prt12), GET_DEV_OBJ(gpio_prt13), GET_DEV_OBJ(gpio_prt14)
```
You could probably utilize `LISTIFY()` macro. Something like:
```
#define GET_DEV_OBJ_FROM_LIST(i, _)
	GET_DEV_OBJ(gpio_prt##i)

LISTIFY(15, GET_DEV_OBJ_FROM_LIST, (,))
```

npal-cy replied:
Thanks! updated.

mniestroj commented on drivers/gpio/gpio_ifx_cat1.c:
```diff
@@ -0,0 +1,315 @@
+
+/* Map port number to device object */
+static const struct device *const port_dev_obj[IOSS_GPIO_GPIO_PORT_NR] = {
+	LISTIFY(15, GET_DEV_OBJ_FROM_LIST, ( , ))
```
```suggestion
	LISTIFY(15, GET_DEV_OBJ_FROM_LIST, (,))
```
this should suppress following warning (and it will make consistent with other code in Zephyr):
```
SPACING
drivers/gpio/gpio_ifx_cat1.c:62  space prohibited after that open parenthesis '('
```

npal-cy replied:
updated

gmarull reviewed (dismissed):
I think major issues have been addressed, should be good to go

npal-cy commented:
Updated RAM/ROM size implementation for CM0p images. 

PSOC6/CAT1 stop works after commit [44628735](https://gitlab.intra.infineon.com/software-integration/zephyrproject-rtos/zephyr/-/commit/44628735b870b2806bbea47477c3300bb624ad31), where was updated rom_start_offset(add to address instead of set.)

The following updates was done to handled  [44628735](https://gitlab.intra.infineon.com/software-integration/zephyrproject-rtos/zephyr/-/commit/44628735b870b2806bbea47477c3300bb624ad31) changes:
- added own hidden SOC_PSOC6_CM0P_IMAGE_ROM/RAM_SIZE kconfig object to handle RAM/ROM size for PSOC6 CM0P images. 
- replaced ROM_START_OFFSET to SOC_PSOC6_CM0P_IMAGE_ROM_SIZE in `soc/arm/infineon_cat1/psoc6/Kconfig.defconfig`
- replaced SRAM_OFFSET to SOC_PSOC6_CM0_IMAGE_RAM_SIZE in `soc/arm/infineon_cat1/psoc6/Kconfig.defconfig`
- updated `soc/arm/infineon_cat1/psoc6/rom_cm0image.ld` and `/soc/arm/infineon_cat1/psoc6/ram_cm0image.ld` to use CONFIG_SOC_PSOC6_CM0P_IMAGE_ROM/RAM_SIZE 

rebased commit: https://github.com/zephyrproject-rtos/zephyr/pull/44211/commits/fca19c19a0fb3e62facc936b2c4c1bf9cba99912

carlescufi reviewed (approved):
The manifest and binary blobs changes LGTM.

carlescufi commented:
The Compliance Failures are just the usual pinctrl typedef.
